### PR TITLE
Matrix build with many Maven versions

### PR DESCRIPTION
**Pull request by slawekjaranowski**
### break compatibility

 - replace `maven_version` by  `maven-matrix` and `ff-maven`
 - remove not used `verify-site-goal`
 - rename `maven_args` to `maven-args`

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

File: .github/workflows/maven-verify-test.yml [hunk 1]
```diff
@@ -32,3 +32,4 @@ jobs:
       ff-goal: test
       verify-goal: clean install
```
Comment (michael-o) [inline .github/workflows/maven-verify-test.yml:33]: Do we really need install? Isn't verify sufficient and saves time?

File: .github/workflows/maven-verify-test.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 name: Verify - Test
 
-on: push
+on: [ push, pull_request ]
```
Comment (olamy) [inline .github/workflows/maven-verify-test.yml:20]: is this change related to Matrix build?

File: .github/workflows/maven-verify.yml [hunk 3]
```diff
@@ -20,54 +20,54 @@ name: Verify
...
         type: number
 
       timeout-minutes:
-        description: 'timeout-minutes used by the builds (defaults to 360)'
+        description: timeout-minutes used by the builds (defaults to 360)
         required: false
         default: 360
         type: number
 
-      jdk-distribution-matrix:
```
Comment (olamy) [inline .github/workflows/maven-verify.yml:65]: is this compatible for builds currently using this argument? (such https://github.com/apache/maven-compiler-plugin/blob/master/.github/workflows/maven-verify.yml)

File: .github/workflows/maven-verify-test.yml (hunk 2 above)
Comment (slawekjaranowski) [reply .github/workflows/maven-verify-test.yml:20]: not 😄

File: .github/workflows/maven-verify-test.yml (hunk 1 above)
Comment (slawekjaranowski) [reply .github/workflows/maven-verify-test.yml:33]: it is only test workflow executed in this repo - show what happens when `ff-goal` and `verify-goal` have different commands

File: .github/workflows/maven-verify-test.yml [hunk 4]
```diff
@@ -32,3 +32,4 @@ jobs:
       ff-goal: test
       verify-goal: clean install
       verify-fail-fast: false
+      maven-matrix: '[ "3.2.5", "3.8.6" ]'
```
[the remark this answers is not in the document]
Comment (slawekjaranowski) [reply .github/workflows/maven-verify-test.yml:35]: t is only test workflow executed in this repo

File: .github/workflows/maven-verify.yml (hunk 3 above)
Comment (slawekjaranowski) [reply .github/workflows/maven-verify.yml:65]: I grupped all `*-matrix` options together - it is preserved in line 41

File: .github/workflows/maven-verify.yml [hunk 5]
```diff
@@ -20,54 +20,54 @@ name: Verify
 on:
   workflow_call:
     inputs:
-      maven_args:
+      maven-args:
```
Comment (olamy) [inline .github/workflows/maven-verify.yml:23]: is this backward compatible?

Comment (slawekjaranowski) [reply .github/workflows/maven-verify.yml:23]: not - description update
so I created next branch - version v3

changed to be consequently in options name format

**Review by olamy — APPROVED** [review]
